Tests/SourceKitLSPTests/WorkspaceTests.swift

```diff
@@ -967,4 +968,171 @@ final class WorkspaceTests: XCTestCase {
       ["Cannot convert value of type 'Int' to specified type 'String'"]
     )
   }
+
+  func testWorkspaceOptionsOverrideBuildSystem() async throws {
+    let sdkArgs =
+      if let defaultSDKPath {
+        """
+        -sdk '\(defaultSDKPath)'
+        """
+      } else {
+        ""
+      }
+
+    let project = try await MultiFileTestProject(files: [
+      ".sourcekit-lsp/config.json": """
+      {
+        "defaultWorkspaceType": "compilationDatabase"
+      }
+      """,
+      "src/Foo.swift": """
+      #if HAVE_SETTINGS
+      #error("Have settings")
+      #endif
+      """,
+      "Sources/MyLib/Bar.swift": "",
+      "build/compile_commands.json": """
+      [
+      {
+        "directory": "$TEST_DIR",
+        "command": "swiftc $TEST_DIR/src/Foo.swift -DHAVE_SETTINGS \(sdkArgs)",
+        "file": "src/Foo.swift",
+        "output": "$TEST_DIR/build/Foo.swift.o"
+      },
```

Comment:
I’m surprised this works. AFAIK JSON doesn’t support trailing commas…

Reply:
TIL JSONDecoder allows trailing commas. I'll remove them in any case, doesn't really matter here.